### PR TITLE
Implement getBalance for Bitcoin

### PR DESCRIPTION
**Pull request by i3wgnit**
### Description

Implement getBalance for BitcoinLedgerProvider
First merge #48 before this one.

### Parts

- [x] Implement `getBalance` method
- [x] Make an interface for the previous bullet point in the `Client`

### CODE REVIEW

File: src/providers/bitcoin/BitcoinLedgerProvider.js [hunk 1]
```diff
@@ -121,6 +121,13 @@ export default class BitcoinLedgerProvider extends Provider {
     return buffer.reverse() // Amount needs to be little endian
   }
 
+  async getBalance (config = { segwit: false }) {
```
Comment (monokh) [inline src/providers/bitcoin/BitcoinLedgerProvider.js]: i find it quite un normalized that the functions accept a config like that... the same interface won't exist for ethereum. The correct way i believe should be to set this flag on the provider constructor, but that's a different issue that doesn't need to be addressed here.

File: src/providers/bitcoin/BitcoinLedgerProvider.js [hunk 2]
```diff
@@ -121,6 +121,13 @@ export default class BitcoinLedgerProvider extends Provider {
...
+    await this._connectToLedger()
+
+    const { usedAddresses } = await this._getSpendingDetails(config)
+    return usedAddresses.reduce((acc, detail) => acc + detail.value, 0)
```
Comment (monokh) [inline src/providers/bitcoin/BitcoinLedgerProvider.js]: This type of thing is a good candidate for a unit test. Mock _connectToLedger and _getSpendingDetails and see if the function does what you expect. Again different issue 🙂 feel free to address it whenever in your priorities

**Review by monokh — APPROVED** [review]
LGTM, minor comments but nothing that can't be done separately

**Review by monokh — CHANGES_REQUESTED** [review]
looks fine but again can you include an example so we can see how to use this? similarly for the ethereum version

File: src/Client.js [hunk 3]
```diff
@@ -284,6 +284,40 @@ export default class Client {
...
+    if (addresses != null) {
+      if (!isArray(addresses)) {
+        throw new TypeError('Addresses should be an array')
+      }
+
+      if (!addresses.every(isString)) {
+        throw new TypeError('All addresses should be strings')
+      }
+
+      if (!addresses.every(addr => (/^[A-Fa-f0-9]+$/.test(addr)))) {
```
Comment (matthewjablack) [inline src/Client.js:307]: Doesn't this need to be removed since bitcoin addresses are not hex strings?

**Pull request merged**